crates/uv-distribution/src/distribution_database.rs

```diff
@@ -194,17 +194,46 @@ impl<'a, Context: BuildContext> DistributionDatabase<'a, Context> {
                         .to_file_path()
                         .map_err(|()| Error::NonFileUrl(url.clone()))?;
                     return self
-                        .load_wheel(&path, &wheel.filename, wheel_entry, dist, hashes)
+                        .load_wheel(
+                            &path,
+                            &wheel.filename,
+                            WheelExtension::Whl,
+                            wheel_entry,
+                            dist,
+                            hashes,
+                        )
                         .await;
                 }
 
+                // Add the `.zst` extension to the URL, if necessary.
```

Comment:
Nit: It's a little funny these are three conditionals instead of, e.g., one returning a tuple

Reply:
Added a struct!